### DISCUSSION TITLE
Hacktoberfest call

### DISCUSSION THREAD

**Discussion by kossnocorp** [js-fns/date-fns#2013] [General]
__TL;DR: date-fns is calling Hacktoberfest participants to help with [TypeScript migration](https://github.com/date-fns/date-fns/discussions/1932). [**How to contribute**](https://github.com/date-fns/date-fns-typescriptify/issues/218).__

In light of the [recent Hacktoberfest controversy](https://blog.domenic.me/hacktoberfest/), I decided to support the initiative and try to demonstrate how such a chaotic force could be directed and turned into something positive.

[Recently, I announced that date-fns is migrating to TypeScript](https://github.com/date-fns/date-fns/discussions/1932). Given the size of the codebase and failed attempts to migrate the whole library at once, I opted for gradual migration.

Despite great efforts applied by many date-fns contributors, we still need to migrate 217 functions and locales. **And this is exactly the type of work Hacktoberfest participants could help with**.

**Let's make an open-source community better together showing a great example of Hacktoberfest done right.**

## How to contribute

1. Head to [the list of functions and locales to migrate](https://github.com/date-fns/date-fns-typescriptify/issues).
2. Find an unclaimed issue (an issue without `⚠ Claimed` label on it) and leave a comment.
3. Fork [the repo](https://github.com/date-fns/date-fns), migrate the claimed function or locale, and send a pull-request. **To prevent stalled work, make sure that you finish the PR in a week, otherwise, the claim will be lifted.**
4. Add `TypeScript migration` label (also add `Hacktoberfest` [if you participate in it](https://hacktoberfest.digitalocean.com/)).
5. Thank you!

**Comment by numToStr** [comment]
Is there any functions or locales which is already migrated to typescript? So, we can take that as a guide to implement types for the remaining ones.

**Reply by kossnocorp** [reply]
Sure, there’re plenty of migrated functions (search index.ts and test.ts), for example https://github.com/date-fns/date-fns/blob/master/src/addDays/index.ts

**Comment by minitesh** [comment]
One of my PR is already pending since so many days.

https://github.com/date-fns/date-fns/pull/1196

Can you please take a look at it so that I can also make some more contributions and its because currently all my changes are in my forked master branch and that is why I need to get to done first.

**Reply by kossnocorp** [reply]
I appreciate the PR, but I can't promise any ETA on merging it. New features take quite some time to review and see how it fits the big picture, and since I have limited time, I take care of low-hanging fruits first. I hope you understand.

Said that I recommend you create another branch and send PRs from it.

No need to create a bottleneck for both you from it and me.

**Reply by minitesh** [reply]
Yes @kossnocorp no problem.
I have found another way and I have raised PRs with that.
So no issues.
Thanks

**Reply by kossnocorp** [reply]
Thank you a lot!
